python/lsst/pipe/base/task.py

```diff
@@ -127,7 +127,7 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
                 config = self.ConfigClass()
             self._taskDict = dict()
             loggerName = self._fullName
-            if log is not None:
+            if log is not None and log.getName():
```

Comment:
Apparently yes. Try building pipe_base without this commit - testCmdLineTask fails with a fairly unhelpful error message.